ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -798,7 +799,17 @@ def set_waypoints_velocity(self):
         accel_ratio = 2.0  # increase slowing down distance
         recalc = False       # indication that JMT calcs exceed bounds
 
-        dist_to_tl = self.get_dist_to_tl()
+        if self.final_waypoints_start_ptr == len(self.waypoints) - 1:
+            if self.got_to_end is False:
+                rospy.logwarn("reached end of track at ptr = {}".format(self.final_waypoints_start_ptr))
+                self.got_to_end = True
+            dist_to_tl = 0.0
+        else:
+            dist_to_tl = self.get_dist_to_tl()
+
+        # don't go beyond end of track
+        if self.final_waypoints_start_ptr + self.lookahead_wps > len(self.waypoints)-1:
+            self.lookahead_wps = (len(self.waypoints)) - self.final_waypoints_start_ptr
```

Comment:
Moved this logic around so that the car will stay stopped when it reaches the end of the track, but we will see that it will continue sending the final waypoint as expected.